src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java

```diff
@@ -0,0 +1,124 @@
+import uk.gov.hmcts.reform.civil.model.*;
+import uk.gov.hmcts.reform.civil.model.documents.CaseDocument;
+import uk.gov.hmcts.reform.civil.model.documents.Document;
+import uk.gov.hmcts.reform.civil.model.documents.DocumentMetaData;
+import uk.gov.hmcts.reform.civil.sampledata.CaseDataBuilder;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.civil.model.documents.DocumentType.SEALED_CLAIM;
+import static uk.gov.hmcts.reform.civil.service.docmosis.DocmosisTemplates.N1;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {
+    CivilDocumentStitchingService.class,
+    JacksonAutoConfiguration.class,
+    CaseDetailsConverter.class
+})
+class CivilDocumentStitchingServiceTest {
+    @Autowired
+    private CivilDocumentStitchingService civilDocumentStitchingService;
+    @MockBean
+    private BundleRequestExecutor bundleRequestExecutor;
+    @MockBean
+    private StitchingConfiguration stitchingConfiguration;
+    private static final CaseDocument CLAIM_FORM =
+        CaseDocument.builder()
+            .createdBy("John")
+            .documentName(String.format(N1.getDocumentTitle(), "000DC001"))
+            .documentSize(0L)
+            .documentType(SEALED_CLAIM)
+            .createdDatetime(LocalDateTime.now())
+            .documentLink(Document.builder()
+                              .documentUrl("fake-url")
+                              .documentFileName("file-name")
+                              .documentBinaryUrl("binary-url")
+                              .build())
+            .build();
+    private List<DocumentMetaData> buildDocumentMetaDataList() {
+        return List.of(
+            new DocumentMetaData(
+                CLAIM_FORM.getDocumentLink(),
+                "Sealed Claim Form",
+                LocalDate.now().toString()
+            )
+        );
+    }
+
+    private CaseData buildCaseData(Bundle bundle){
+        List<IdValue<Bundle>> caseBundles = List.of(new IdValue<>("1", bundle));
+        return CaseDataBuilder.builder().caseBundles(caseBundles).build();
+    }
+    @Test
+     void whenCaseDataIsNullWithPayloadAndNoCaseBundlesNoStitchingDocument() {
+        //Given: Payload details but case data not provided
+        List<DocumentMetaData> documentMetaDataList = new ArrayList<>();
+        CaseData caseData = CaseDataBuilder.builder().legacyCaseReference("ClaimNumber").build();
+        //Then: the case documents is null
+        CaseDocument caseDocument = civilDocumentStitchingService.bundle(documentMetaDataList, "Auth",
+                                                                  "Title", "FileName", caseData);
+        Assertions.assertNull(caseDocument);
```

Comment:
It's missing the `// When` section.

```suggestion
        //When: I try to bundle the document
        CaseDocument caseDocument = civilDocumentStitchingService.bundle(documentMetaDataList, "Auth",
                                                                  "Title", "FileName", caseData);
        //Then: the case documents is null
        Assertions.assertNull(caseDocument);
```